notification/src/handler/notification/notification.handler.js

```diff
@@ -200,7 +213,21 @@ async function calculateUpdateActionsForPayment(payment, notification, logger) {
         )
       )
     }
+    const paymentKey = payment.key
+
+    if (
+      notificationRequestItem.eventCode === 'AUTHORISATION' &&
+      (notificationRequestItem.success === "true" || notificationRequestItem.success === true) &&
+      pspReference !== paymentKey &&
+      oldTransaction?.state !== 'Success'
```

Comment:
I wonder why we care about success state, state of the oldTransaction or eventCode type? Maybe it is just enough to check (same as we do above for payment fetching) if originalReference or pspReference is set. Something like that:

```
const newPspReference = notification.originalReference?originalReference:notification.pspReference
if (newPspReference && newPspReference != payment.key) {
   updateActions.push({
          action: 'setKey',
          key: newPspReference,
    })
}
```

Reply:
Because it adds extra update action if it is not authorization notification. For cases like refund, cancel payment. I suppose the payment has already used pspReference as key. I don't need to update it again with original reference.

Reply:
Anyway I updated the code
https://github.com/commercetools/commercetools-adyen-integration/pull/1049/commits/377900be32230fd4d786dbebfb4159a0ee7eae1e